src/test/steps/apiTestSteps.js

```diff
@@ -0,0 +1,154 @@
+let _ = require('lodash');
+
+function getPostBody(userIdKey, valueUserId, titleKey, randomPostTitle, bodyKey, randomPostBody){
+  let bodyWithUserIdTitleBody = {
+    [`${userIdKey}`]: valueUserId,
+    [`${titleKey}`]: randomPostTitle,
+    [`${bodyKey}`]: randomPostBody,
+  }
+  return bodyWithUserIdTitleBody;
+};
+
+$baseURL = new APIUtils("https://jsonplaceholder.typicode.com");
+
+When(/^the user send get request to '(.*)'$/, async function (endpoint) {
+    const response = await $baseURL.getRequest(endpoint);
+    this.sharedStatusCode = response.status;
+    this.responseData = response.data;
+    this.responseSize = this.responseData.length;
+  });
+
+  Then(/^the API status code should be 200 - OK$/, function () {
+    expectChai(this.sharedStatusCode).to.equal(
+      StatusCodes.OK,
+      `The status code should be correct. \r\nResponse body:\r\n ${JSON.stringify(
+        this.sharedResponseBody
+      )}`
+    );
+  });
+
+  Then(
+    /^the response is in json format$/,
+    function () {
+      expectChai(this.responseData).to.be.jsonObj();
+    }
+  );
+
+  Then(
+    /^the response posts are sorted ascending by id$/,
+    function () {
+      let sortedAscendingById = true;
+      for(let i = 2; i < this.responseSize; i++){
+        if(this.responseData[i].id < this.responseData[i-1].id){
+          sortedAscendingById = false;
+          break;
+        }
+      }
+      expectChai(sortedAscendingById).to.be.true;
+    }
+  );
+
+  When(/^the user send get request to '(.*)' with '(.*)' id$/, async function (endpoint, id) {
+    const requestUrl = endpoint+"/"+id;
+    const response = await $baseURL.getRequest(requestUrl);
+    this.sharedStatusCode = response.status;
+    this.responseData = response.data;
+    this.sentId = id;
+  });
+
+  Then(/^the API status code should be 201 - Created$/, function () {
+    expectChai(this.sharedStatusCode).to.equal(
+      StatusCodes.CREATED,
+      `The status code should be correct. \r\nResponse body:\r\n ${JSON.stringify(
+        this.sharedResponseBody
+      )}`
+    );
+  });
+
+  Then(
+    /^the post id is match '(.*)'$/,
+    async function (endpointNumber) {
+      const expectedId = expectedPostsData[endpointNumber-1].id;
+      expectChai(this.responseData.id).to.equal(expectedId);
+    }
+  );
+
+  Then(
+    /^the post userID is match '(.*)'$/,
+    async function (expectedUserId) {
+      expectChai(this.responseData.userId).to.equal(expectedUserId);
+
+    }
+  );
+
+  Then(
+    /^the '(.*)' is not empty$/,
+    async function (bodyKeyName) {
+      expectChai(this.responseData[`${bodyKeyName}`]).to.not.equal("");
+    }
+  );
+
+  When(/^the user send post request to '(.*)' with '(.*)' equals '(.*)', '(.*)' equals '(.*)', '(.*)' equals '(.*)'$/, 
+  async function (endpoint, userIdKey, userIdValue, titleKey, titleValue, bodyKey, bodyValue) {
+    let postRequestBody = getPostBody(userIdKey, userIdValue, titleKey, titleValue, bodyKey, bodyValue);
```

Comment:
here we can use data from testData files instead of passing all parameters into the step, for example you can have a body with random values to check post request, then step will be called something like the user send post request to '(.*)' with random values, what if our post request will have 50-100 fields? we won't pass them all as parameters to step